### PR TITLE
Fix VBox-Loader

### PR DESCRIPTION
**Pull request by cecinestpasunepipe**
Fix VBox Loader and use XPath instead of manually walking over XML-nodes.
Remove snapshots() method as these are not supported yet.

(DIS-1430)

### CODE REVIEW

File: dissect/hypervisor/disk/vdi.py [hunk 1]
```diff
@@ -67,18 +67,13 @@ def _read(self, offset, length):
...
-        self.fh = fh
-        self.xml = ElementTree.fromstring(fh.read())
 
-        self.machine = list(self.xml)[0]
-        self.registry = next((elem for elem in self.machine if "MediaRegistry" in elem.tag))
+    VBOX_XML_NAMESPACE = "{http://www.virtualbox.org/}"
+
+    def __init__(self, fh):
+        xml = ElementTree.fromstring(fh.read())
+        self._disks = xml.findall(f".//{self.VBOX_XML_NAMESPACE}HardDisk[@location][@format='VDI'][@type='Normal']")
```
Comment (Schamper) [inline dissect/hypervisor/disk/vdi.py:75]:
```suggestion
        self.xml = ElementTree.fromstring(fh.read())
```

File: tests/test_vbox.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+                        <MediaRegistry>
+                            <HardDisks>
+                                <HardDisk location="os2warp4.vdi" format="VDI" type="Normal" />
+                            </HardDisks>
+                        </MediaRegistry>
+                    </Machine>
+                </VirtualBox>
+    """
+
+    with StringIO(xml) as fh:
```
Comment (Schamper) [inline tests/test_vbox.py:20]:
```suggestion
    xml = """
    <?xml version="1.0"?>
    <VirtualBox xmlns="http://www.virtualbox.org/">
        <Machine>
            <MediaRegistry>
                <HardDisks>
                    <HardDisk location="os2warp4.vdi" format="VDI" type="Normal" />
                </HardDisks>
            </MediaRegistry>
        </Machine>
    </VirtualBox>
    """

    with StringIO(xml.strip()) as fh:
```
Maybe personal preference but looks a bit nicer :)

File: dissect/hypervisor/disk/vdi.py (hunk 1 above)
Comment (cecinestpasunepipe) [reply dissect/hypervisor/disk/vdi.py:75]: Why would you expose the xml as a public property?

File: dissect/hypervisor/disk/vdi.py [hunk 3]
```diff
@@ -67,18 +67,13 @@ def _read(self, offset, length):
...
 
     def disks(self):
-        for hdd_elem in next((elem for elem in self.registry if "HardDisks" in elem.tag)):
+        for hdd_elem in self._disks:
```
[the remark this answers is not in the document]
Comment (cecinestpasunepipe) [reply dissect/hypervisor/disk/vdi.py:78]: I do not understand the rationale behind this suggestion.

File: tests/test_vbox.py (hunk 2 above)
Comment (cecinestpasunepipe) [reply tests/test_vbox.py:20]: That will make the xml invalid and the test will fail.

File: dissect/hypervisor/disk/vdi.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (cecinestpasunepipe) [reply dissect/hypervisor/disk/vdi.py:78]: Why would you expose the xml as a public property?

File: dissect/hypervisor/disk/vdi.py (hunk 1 above)
Comment (cecinestpasunepipe) [reply dissect/hypervisor/disk/vdi.py:75]: I do not understand the rationale behind this suggestion. Can you elaborate?

File: tests/test_vbox.py (hunk 2 above)
Comment (cecinestpasunepipe) [reply tests/test_vbox.py:20]: That will make the xml invalid and the test will fail.

Comment (Schamper) [reply tests/test_vbox.py:20]: The `.strip()` will take care of that right?

File: dissect/hypervisor/disk/vdi.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Schamper) [reply dissect/hypervisor/disk/vdi.py:78]: `._xml` in that case, mostly just moving the pre-reading of the disks out of the constructor.

File: dissect/hypervisor/disk/vdi.py (hunk 1 above)
Comment (Schamper) [reply dissect/hypervisor/disk/vdi.py:75]: It's neater to not do the pre-reading of the disk attributes in the constructor, if you're not going to use them in the constructor. IMO it makes more sense to only start reading the disk attributes in the actual `disks()` method.

File: dissect/hypervisor/disk/vdi.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Schamper) [reply dissect/hypervisor/disk/vdi.py:78]: Also if at a later stage you'll add `snapshots()` or something else, or some other piece of code may want to inspect the XML, it's available.

File: dissect/hypervisor/disk/vdi.py [hunk 4]
```diff
@@ -67,18 +67,12 @@ def _read(self, offset, length):
 
 
 class Vbox:
```
Comment (Schamper) [inline dissect/hypervisor/disk/vdi.py:69]: Do you think it's worth moving this class to `dissect/hypervisor/descriptor/vbox.py` with this MR? Would fit in better with the rest there.

Comment (cecinestpasunepipe) [reply dissect/hypervisor/disk/vdi.py:69]: Yes, that makes sense, good idea.

Comment (Schamper) [inline dissect/hypervisor/disk/vdi.py:69]: Remove the old class.

File: tests/test_vbox.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
+from io import StringIO
+
+from dissect.hypervisor.disk.vdi import Vbox
```
Comment (Schamper) [inline tests/test_vbox.py:3]: Update this import.

**Review by Schamper — CHANGES_REQUESTED** [review]

**Review by Schamper — APPROVED** [review]

**Pull request merged**